opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -658,6 +661,16 @@ diskLoop:
 			imageID, _ := disk.GetI(shared.ImageID)
 			diskMap["image_id"] = imageID
 
+			// for volatile disk, TYPE has the same value
+			// than DISK_TYPE
+			if imageID == -1 {
+				diskType, _ := disk.Get("TYPE")
```

Comment:
In case of Volatile disk, the type of the volatile disk is stored under `TYPE` in OpenNebula ?